frontend/src/components/indices/IndicesValidate.js

```diff
@@ -18,27 +18,31 @@ import PageContent from "../PageContent";
 import IndicesValidationResult from "./IndicesValidationResult";
 import * as Options from "../../utils/options";
 import api, { withToken } from "../../utils/api";
-import { validate, list } from "../../modules/indices/actions";
+import { list, validate } from "../../modules/indices/actions";
 import { requiredRules } from "../../constants";
-import store from "../../store";
+import { useAppDispatch, useAppSelector } from "../../hooks";
+import { selectToken } from "../../selectors";
 
 // API functions
 const listSets = (token, options) =>
   withToken(token, api.indices.listSets)(options).then(res => res.data)
 
-const listInstrumentTypes = () => store.dispatch(api.instrumentTypes.list({}))
-
+const IndicesValidate = () => {
+  const history = useNavigate();
 
-const mapStateToProps = state => ({
-  token: state.auth.tokens.access,
-  indicesTotalCount: state.indices.totalCount,
-  isFetching: state.indices.isFetching,
-});
+  const dispatch = useAppDispatch()
 
-const actionCreators = { list, validate };
+  const listCallback = useCallback((options) => {
+    return dispatch(list(options))
+  }, [dispatch])
+  const validateCallback = useCallback((options) => {
+    return dispatch(validate(options))
+  }, [dispatch])
+  const listInstrumentTypes = useCallback(() => {
+    return dispatch(api.instrumentTypes.list({ instruments__isnull: false }))
```

Comment:
This returns one instrument_type for each instrument... This means you have duplicates in the dropdown in the frontend... You could try adding .distinct() to the queryset in the viewset. If it does not work you may have to resolve the duplications in the frontend.